tests/test_decision_maker/test_base.py

```diff
@@ -452,8 +539,91 @@ def test__is_utility_enhancing(self):
         self.decision_maker.ownership_state._quantities_by_good_id = None
         assert self.decision_maker._is_utility_enhancing(tx_message)
 
+    def test_sign_tx(self):
```

Comment:
see below, please split up into one method per assert and clearly describe what is being tested